modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/PerformaceStatisticsProcessor.java

```diff
@@ -0,0 +1,232 @@
+
+    /** Synchronization mutex for request futures. */
+    private final Object mux = new Object();
+
+    /** Enable/disable statistics request futures. */
+    private final ConcurrentMap<UUID, GridFutureAdapter<Void>> reqFuts = new ConcurrentHashMap<>();
+
+    /** Disconnected flag. */
+    private volatile boolean disconnected;
+
+    /** Stopped flag. */
+    private volatile boolean stopped;
+
+    /** @param ctx Kernal context. */
+    public PerformaceStatisticsProcessor(GridKernalContext ctx) {
+        super(ctx);
+
+        writer = new FilePerformanceStatistics(ctx);
+
+        proc = new DistributedProcess<>(ctx, PERFORMANCE_STATISTICS, start -> {
+            if (start) {
+                return ctx.closure().callLocalSafe(() -> {
+                    if (start)
+                        writer.start();
+
+                    return true;
+                });
+            }
+
+            return writer.stop().chain(f -> true);
+        }, (uuid, res, err) -> {
+            if (!F.isEmpty(err) && statisticsEnabled())
+                writer.stop();
+
+            synchronized (mux) {
+                GridFutureAdapter<Void> fut = reqFuts.get(uuid);
+
+                if (fut != null) {
+                    if (!F.isEmpty(err))
+                        fut.onDone(new IgniteException("Unable to process request [err=" + err + ']'));
+                    else
+                        fut.onDone();
+                }
+            }
+        });
+    }
+
+    /** @return Performance statistics writer. */
+    public IgnitePerformanceStatistics writer() {
+        return writer;
+    }
+
+    /** @return {@code True} if collecting performance statistics is enabled. */
+    public boolean statisticsEnabled() {
+        return writer.performanceStatisticsEnabled();
+    }
+
+    /**
+     * Starts collecting performance statistics.
+     *
+     * @return Future to be completed on collecting started.
+     */
+    public IgniteInternalFuture<Void> startStatistics() {
+        if (!allNodesSupports(ctx.discovery().allNodes(), IgniteFeatures.PERFORMANCE_STATISTICS)) {
+            return new GridFinishedFuture<>(
+                new IllegalStateException("Not all nodes in the cluster support collecting performance statistics."));
+        }
+
+        GridFutureAdapter<Void> fut = new GridFutureAdapter<>();
+
+        UUID uuid = UUID.randomUUID();
+
+        synchronized (mux) {
+            if (disconnected || stopped) {
+                return new GridFinishedFuture<>(
+                    new IgniteFutureCancelledException("Node " + (stopped ? "stopped" : "disconnected")));
+            }
+
+            reqFuts.put(uuid, fut);
+        }
+
+        proc.start(uuid, true);
+
+        return fut;
+    }
+
+    /**
+     * Stops collecting performance statistics.
+     *
+     * @return Future to be completed on collecting stopped.
+     */
+    public IgniteInternalFuture<Void> stopStatistics() {
```

Comment:
Typo: let's rename it to `stopCollectStatistics()`

Reply:
Fixed.